### PR TITLE
Store on .com: Enable auto data updates

### LINKED ISSUE #17862: Store: Automatically run WooCommerce DB updates
WC 3.2.0 introduces a filter so we can enable automatically run DB updates for Atomic sites (since we don't send them to wp-admin, and they wouldn't see notices). We need a hotfix that sets the filter to true.

See https://github.com/woocommerce/woocommerce/pull/16711. Also see the thread at p90Yrv-8K-p2

### PR DESCRIPTION
**Pull request by emdashcodes**
This leverages the filter added in https://github.com/woocommerce/woocommerce/pull/16711, so that we can enable data updates automatically, without our Store users needing to go into wp-admin to click the notice.

Fixes https://github.com/Automattic/wp-calypso/issues/17862.

To Test:
* Install this branch
* Switch your WooCommerce install to an older install. I did `git checkout tags/3.1.2`
* Add a debug line somewhere so that you can monitor the db version and make sure it updates. I added this to the hotfix file above the filter. `error_log( get_option( 'woocommerce_db_version', null ) );`
* Switch your WooCommerce install to 3.2. 
* Visit wp-admin. You should NOT see a notice about db updates. Your debug line should show the db version switched to 3.2.0.
* You can downgrade back to 3.1.2. Now, instead of visting `wp-admin`, make a REST API request. The next request you make after that should show you that the db version correctly switched to 3.2.0.

### CODE REVIEW

**Comment by ryelle** [comment]
I just walked through the testing steps, and it works perfectly through wp-admin. Through the REST API, though, I see a lag – I have to hit an endpoint a few times before the update triggers. It does eventually go through, but I'm not sure why it would delay. It seems like the code isn't hitting the `woocommerce_enable_auto_update_db` filter.

I thought maybe it was the `wc_installing` transient, but that is being cleared after update.

**Comment by allendav** [comment]
I noticed this filter was added to woocommerce core as well - should we be taking advantage of that as well, either in this PR or another?

https://github.com/woocommerce/woocommerce/pull/16711/files#diff-8fe338fa3821970099766ecf56c6fa58R239

**Comment by allendav** [comment]
Tests well for me (thank you for providing the detailed testing hints), but like @ryelle I too noticed a quirk. My error logging (on init) consistently showed the old version exactly once after the newer version. Subsequent REST API calls (or page hits) consistently showed the new version, i.e.:

```
[Tue Oct 10 11:14:58.210927 2017] [:error] [pid 26342] [client ::1:63695] woocommerce_db_version: 3.1.2, referer: http://localhost/wc31_us_2/wp-cron.php?doing_wp_cron=1507659297.9177069664001464843750
[Tue Oct 10 11:14:58.239304 2017] [:error] [pid 26339] [client ::1:63693] woocommerce_db_version: 3.1.2, referer: http://localhost/wc31_us_2/wp-admin/admin-ajax.php?action=wp_wc_updater&nonce=f07a169f95
[Tue Oct 10 11:14:58.597813 2017] [:error] [pid 341] [client ::1:63697] woocommerce_db_version: 3.2.0, referer: http://localhost/wc31_us_2/wp-cron.php?doing_wp_cron=1507659298.3480310440063476562500
[Tue Oct 10 11:14:58.621169 2017] [:error] [pid 26232] [client ::1:63691] woocommerce_db_version: 3.1.2

[Tue Oct 10 11:15:14.091300 2017] [:error] [pid 26338] [client ::1:63699] woocommerce_db_version: 3.2.0
[Tue Oct 10 11:15:18.810746 2017] [:error] [pid 26338] [client ::1:63699] woocommerce_db_version: 3.2.0
[Tue Oct 10 11:15:21.399315 2017] [:error] [pid 26338] [client ::1:63699] woocommerce_db_version: 3.2.0
[Tue Oct 10 11:15:23.374612 2017] [:error] [pid 26338] [client ::1:63699] woocommerce_db_version: 3.2.0
```

That said, I think its good to go nonetheless. :shipit:

**Review by allendav — APPROVED** [review]

**Comment by emdashcodes** [comment]
> I noticed this filter was added to woocommerce core as well - should we be taking advantage of that as well, either in this PR or another?

I started looking at including that in this PR, but I'm not actually sure we should. If they go through our setup flow, the new logic here would make it so the notice won't display in `wp-admin` anymore. And on the other hand, if someone manually installs WooCommerce themselves and is managing it in the wp-admin, we also don't want to block them from setting things up.

**Comment by emdashcodes** [comment]
The upgrade routine actually gets queued and runs in the background so that should explain why there is a slight delay.
